### DISCUSSION TITLE
Outputing eam.i files at the same frequency as eam.r

### DISCUSSION THREAD

**Discussion by alicebarthel** [E3SM-Project/E3SM#5409] [Development discussion]
Hi, 

I am working on a hybrid E3SM run, with a start date of April 1st. 
@golaz helpfully provided monthly restarts from the v2.1 piControl to use as starting points for various components.
As far as I understand, "hybrid" runs require eam.i files, which are only outputed on Jan 1st. (Other components are fine with .r. or .rst. files). 
Is there a good way to use the April 1st data for a hybrid run? (eam.i and eam.r have different ncol_d dimensions)
I wonder if **we could set the eam.i output frequency to match the same frequency as eam.r**? This would give us the flexibility to use hybrid at any point that we have restarts for... 
If there is already a way to do this, let me know!

**Comment by mt5555** [comment]
That's controlled by the inithist namelist variable.  Default is YEARLY, but can be DAILY or MONTHLY (and maybe other options).

You might be able to set it to something really short, like HOURLY, and then do a restart run to get a .i just after the restart file.

**Reply by alicebarthel** [reply]
OK, thanks Mark, that's great to know! So we should just remember to modify the inithist along with REST_N to get both eam.i and eam.r in the future. I may run a very short run with HOURLY as you suggested to double check. Thanks again!

**Reply by golaz** [reply]
Thanks, @mt5555. I wasn't aware of that option. I should have set it to MONTHLY when doing the reruns.

Would it be possible to have EAM automatically save `eam.i` files when it saves restart files? Or alternatively, can you do a hybrid restart using `eam.r` instead of `eam.i` files?

**Comment by tangq** [comment]
FYI. The frequency of `inithist` isn't as flexible as `REST_N`. I believe the shortest `inithist` option is "6-HOURLY".

**Reply by alicebarthel** [reply]
Thanks, that's good to know.

**Comment by mt5555** [comment]
One other suggestion:  could you do a branch run instead of a hybrid run?   I think a branch run will start from all the .r files.

**Reply by alicebarthel** [reply]
My configuration is a textbook "hybrid" (mix-and-matching initial conditions, resetting the year to 0001 to start a long spin-up, etc.), so I was keen to use the most appropriate infrastructure available. Technically, I probably could fool the model by over-writing the values for the ocean and sea ice initial conditions in the restart files and run a branch run (since I am using the same piControl config), but it feels like a less reproducible/documented methodology. 
I appreciate the suggestion for alternatives though. 
At this stage, it may be easiest to re-run 3 months of the piControl with inithist= MONTHLY, then have the hybrid start in April.
